docs/theming.md

```diff
@@ -0,0 +1,356 @@
+# MDL V2 Theming Guide
+
+## Overview
+
+MDL includes a theming module that has been designed for making it easy to change the colors on your application, at
+multiple levels on your toolchain. At the moment, MDL supports theming at the Sass and CSS Custom Property levels, with
+plans to support it at the CDN level as well, once that service is available.
+
+## Colors
+
+MDL theming, like Material Design theming, uses two main colors: _primary_ and _accent_. The primary color is used
+throughout most of the application and components, as the main color for your application. The accent color is used
+for floating action buttons and other interactive elements, serving as visual contrast to the primary.
+
+In addition to the primary and contrast colors, MDL also defines a _background_ color, which is used as a background in
+components, and usually as the page background as well.
+
+Finally, MDL has a number of text colors, which are used for rendering text and other shapes on top of the primary,
+accent and background colors. These need to be picked to either be dark or light, in order to provide sufficient
+contrast to what's behind them, and have different levels of opacity depending on usage:
+- Primary, used for most text.
+- Secondary, used for text which is lower in the visual hierarchy.
+- Hint, used for text hints (such as those in text fields and labels).
+- Disabled, used for text in disabled components and content.
+- Icon, used for icons.
+
+## Building a themed application
+
+Let's start with a simple application, which displays several cards for different categories. We ultimately want each
```

Comment:
Added [a plunker](https://plnkr.co/edit/qxt7qpPsXgkt9UbMnE36?p=preview). I don't think we should cover the getting started bits in this guide; I'm sending folks over to the getting started guide for that.